sdk/tests/core/CMakeLists.txt

```diff
@@ -11,7 +11,14 @@ include(AddCMockaTest)
 
 # -ld link option is only available for gcc
 if(UNIT_TESTING_MOCKS)
-    set(WRAP_FUNCTIONS "-Wl,--wrap=az_platform_clock_msec -Wl,--wrap=az_platform_sleep_msec")
+    set(WRAP_FUNCTIONS "-Wl,--wrap=az_platform_clock_msec \
```

Comment:
Why do we need these `--wrap` statements?

Reply:
As discussed, we're sticking with GCC mock functions, they require these --wrap statements to be linked